packages/cli/src/commands/validator/status.ts

```diff
@@ -0,0 +1,113 @@
+
+  static flags = {
+    ...BaseCommand.flags,
+    signer: Flags.address({
+      description: 'address of the validator to check if elected and validating',
+    }),
+    validator: Flags.address({
+      description: 'address of the signer to check if elected and validating',
+    }),
+    lookback: flags.integer({
+      description: 'how many blocks to look back for signer activity',
+      default: 100,
+    }),
+  }
+
+  static examples = [
+    'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+  ]
+
+  requireSynced = true
+
+  async run() {
+    const res = this.parse(ValidatorStatus)
+
+    // Check that the specified validator or signer meets the validator requirements.
+    const checker = newCheckBuilder(this, res.flags.signer)
+    if (res.flags.validator) {
+      const account = res.flags.validator
+      checker.isValidator(account).meetsValidatorBalanceRequirements(account)
+    } else if (res.flags.signer) {
+      checker.signerMeetsValidatorBalanceRequirements().signerAccountIsValidator()
+    } else {
+      this.error('Either validator or signer must be specified')
+    }
+    await checker.runChecks()
+
+    // Assign and verify the signer.
+    let signer: Address
+    if (res.flags.signer) {
+      signer = res.flags.signer
+      if (res.flags.validator) {
+        const accounts = await this.kit.contracts.getAccounts()
+        if ((await accounts.signerToAccount(signer)) !== res.flags.validator) {
+          this.error(
+            `Signer ${signer} has never been authorized for account ${res.flags.validator}`
+          )
+        }
+      }
+    } else {
+      const accounts = await this.kit.contracts.getAccounts()
+      signer = await accounts.getValidatorSigner(res.flags.validator!)
+      console.info(`Identified ${signer} as the authorized validator signer`)
+    }
+
+    // Determine if the signer is elected, and get their index in the validator set.
+    const election = await this.kit.contracts.getElection()
+    const signers = await election.getCurrentValidatorSigners()
+    const signerIndex = signers.map((a) => eqAddress(a, signer)).indexOf(true)
+    if (signerIndex < 0) {
+      // Determine whether the signer will be elected at the next epoch to provide a helpful error.
+      const frontrunners = await election.electValidatorSigners()
+      if (frontrunners.some((a) => eqAddress(a, signer))) {
+        this.error(
+          `Signer ${signer} is not elected for this epoch, but is currently winning in the upcoming election. Wait for the next epoch.`
+        )
+      } else {
+        this.error(
+          `Signer ${signer} is not elected for this epoch, and is not currently winning the upcoming election.`
```

Comment:
`and would not be elected to the validator set for the next epoch if an election were to be held now.`

Reply:
👍